File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
      //moore voting algo
      int n=nums.size();
      int candidate=0,vote=0;
      for(int i=0;i<n;i++)
      {
        if(vote==0){
            vote=1;
            candidate=nums[i];
        }
        else{
            if(candidate==nums[i]){
                vote++;
            }
            else{
                vote--;
            }
        }
      }
      return candidate;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:

        n=len(nums)
        candidate=0 
        vote=0
        for i in range(len(nums)):
           if (vote==0):
               vote=1
               candidate=nums[i]
            
           else:
               if(nums[i]==candidate):
                   vote+=1
               else:
                   vote-=1
        
        return candidate
        